### PR TITLE
Add field for capturing working patterns for part time jobs

### PR DESCRIPTION
**Pull request by C-gyorfi**
## Context

We want to add another field to work history because:
- As a provider, I need detail about a candidate’s work history so that I have enough information to make decisions on their application.
- As a candidate, I need to be able to describe my part-time jobs, so that I can show how much work I did.

## Changes proposed in this pull request

This PR updates:
- The `WorkExperienceForm`, `WorkHistory::EditController` and work_history/edit views to capture and display working pattern information when a candidate fills up a part-time work experience record. 
- The `SingleApplicationPresenter` to present this information in `work_experience#description`

### Before
image
image

### After
image
image

## Guidance to review

As discussed with @duncanjbrown in the current API documentation the `The WorkExperience object` has a `description` field, if we include working pattern information in this field, no further API update needed. Let me know if you disagree.
image

## Link to Trello card

https://trello.com/c/M9ktUBA0/861-add-field-for-capturing-working-patterns-for-part-time-jobs

## Things to check
- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/components/work_history_review_component.rb [hunk 1]
```diff
@@ -65,10 +65,10 @@ def job_row(work)
     }
   end
 
-  def type_row(work)
+  def working_pattern_row(work)
     {
-      key: 'Type',
-      value: work.commitment.dasherize.humanize,
+      key: 'Working Pattern',
```
Comment (fofr) [inline app/components/work_history_review_component.rb]:
```suggestion
      key: 'Working pattern',
```

File: app/presenters/vendor_api/single_application_presenter.rb [hunk 2]
```diff
@@ -118,10 +118,16 @@ def experience_to_hash(experience)
...
         commitment: experience.commitment,
-        description: experience.details,
+        description: experience_description(experience),
       }
     end
 
+    def experience_description(experience)
+      return experience.details if experience.working_pattern.blank?
+
+      "#{experience.details} - #{experience.working_pattern}"
```
Comment (fofr) [inline app/presenters/vendor_api/single_application_presenter.rb]: I think the working pattern should come first, and that our concatenation of the two should use line breaks and labels with colons. Appending with a dash at the end will mean this information merges into the details, and could read poorly depending on what's input.

eg:
```
"Working pattern: #{experience.working_pattern}\n\nDescription: #{experience.details}"
```

Comment (C-gyorfi) [reply app/presenters/vendor_api/single_application_presenter.rb]: I think it reads much better than the dash, thanks for you suggestion, updated! 👍

File: app/components/work_history_review_component.rb (hunk 1 above)
Comment (C-gyorfi) [reply app/components/work_history_review_component.rb]: Good catch 👍

File: app/presenters/vendor_api/single_application_presenter.rb (hunk 2 above)
Comment (duncanjbrown) [reply app/presenters/vendor_api/single_application_presenter.rb]: Looks good to me @fofr!

File: app/models/candidate_interface/work_experience_form.rb [hunk 3]
```diff
@@ -22,7 +22,7 @@ class WorkExperienceForm
     validates :role, :organisation,
               length: { maximum: 60 }
 
-    validates :details,
+    validates :details, :working_pattern,
```
Comment (duncanjbrown) [inline app/models/candidate_interface/work_experience_form.rb]: Could we add a better validation message for when it appears at the top of the page?

Screenshot 2020-02-05 at 10 09 50

File: app/components/work_history_review_component.rb [hunk 4]
```diff
@@ -114,4 +114,10 @@ def any_jobs_with_same_role_and_organisation?(work)
     jobs = @application_form.application_work_experiences.where(role: work.role, organisation: work.organisation)
     jobs.many?
   end
+
+  def working_pattern(work)
+    return work.commitment.dasherize.humanize if work.working_pattern.blank?
```
Comment (duncanjbrown) [inline app/components/work_history_review_component.rb]: 📐 👍

File: spec/support/test_helpers/candidate_helper.rb [hunk 5]
```diff
@@ -180,7 +180,9 @@ def candidate_fills_in_work_experience
     with_options scope: 'application_form.work_history' do |locale|
       fill_in locale.t('role.label'), with: 'Teacher'
       fill_in locale.t('organisation.label'), with: 'Oakleaf Primary School'
-      choose 'Full-time'
+      choose 'Part-time'
+
+      fill_in 'Give details about your working pattern', with: 'I had a working pattern'
```
Comment (duncanjbrown) [inline spec/support/test_helpers/candidate_helper.rb]: :+1:

File: app/presenters/vendor_api/single_application_presenter.rb (hunk 2 above)
Comment (C-gyorfi) [reply app/presenters/vendor_api/single_application_presenter.rb]: @duncanjbrown are you happy with the rest of the code? I just want to make sure someone had a second look before I merge.

File: app/models/candidate_interface/work_experience_form.rb (hunk 3 above)
Comment (fofr) [reply app/models/candidate_interface/work_experience_form.rb]: Good catch

Comment (C-gyorfi) [reply app/models/candidate_interface/work_experience_form.rb]: image

@fofr @duncanjbrown Are you happy with this? following the same pattern that skills and experience has.

File: config/locales/application_form.yml [hunk 6]
```diff
@@ -482,6 +482,8 @@ en:
             end_date:
               invalid: Enter an end date in the correct format
               in_the_future: Enter an end date that is not in the future
+            working_pattern:
+              too_many_words: Working pattern must be %{count} words or fewer
```
Comment (fofr) [inline config/locales/application_form.yml]:
```suggestion
              too_many_words: Details about your working pattern must be %{count} words or fewer
```

File: app/models/candidate_interface/work_experience_form.rb (hunk 3 above)
Comment (fofr) [reply app/models/candidate_interface/work_experience_form.rb]: I've add a small content tweak

**Pull request merged**